### PR TITLE
APP-4097 - Toast component

### PR DESCRIPTION
**Pull request by axeleriksson147**
- https://perzoinc.atlassian.net/wiki/spaces/DevX/pages/1930300829/Toast - Proposal
- https://perzoinc.atlassian.net/browse/APP-4097 - Ticket
- https://www.figma.com/file/V0VKw74594EyCKHwqYk6LV/%E2%9D%96-SDS-Component-Library - Design for Toast component

Screenshot 2021-04-30 at 12 13 20
Screenshot 2021-05-04 at 13 02 06

### CODE REVIEW

**Review by swasunb — APPROVED** [review]

File: src/components/atoms/_toast.scss [hunk 1]
```diff
@@ -0,0 +1,27 @@
+$fontSize: 14;
+
+.tk-toast {
+    background-color: rgba(39, 41, 44, 1);
```
Comment (symphony-adnane) [inline src/components/atoms/_toast.scss:4]: hi @axeleriksson147 
All colors should come from for accesors map and be adapted for light and dark theme.
hardcoded colors couldn't be supported

File: src/components/atoms/_toast.scss [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+    border-radius: 16px;
+    color: rgba(241, 241, 243, 1);
+    line-height: toEm(20, $fontSize);
```
Comment (symphony-adnane) [inline src/components/atoms/_toast.scss:7]: let's use directly toRem

File: src/components/atoms/_toast.scss [hunk 3]
```diff
@@ -0,0 +1,27 @@
+$fontSize: 14;
```
Comment (symphony-adnane) [inline src/components/atoms/_toast.scss:1]: you wont' need this when using toRem ;)

File: src/components/atoms/_toast.scss [hunk 4]
```diff
@@ -0,0 +1,27 @@
...
+        color: rgba(176, 179, 186, 1);
+        margin-right: toEm(16, $fontSize);
+    }
+
+    &__icon-right {
+        color: rgba(176, 179, 186, 1);
+        margin-left: toEm(16, $fontSize);
+    }
+
+    &__icon-warning {
```
Comment (symphony-adnane) [inline src/components/atoms/_toast.scss:24]: I don't think the coloring of the icon should be part of this component
IMO what we could do though
is add a variant of toast. 
So for ex: 
tk-toast tk-toast--warning 
Would give the left icon a warning state

File: src/components/atoms/_toast.scss (hunk 1 above)
Comment (axeleriksson147) [reply src/components/atoms/_toast.scss:4]: Ops, I placed this todo in my head and forgot it... Will look into and fix...

File: src/components/atoms/_toast.scss (hunk 2 above)
Comment (axeleriksson147) [reply src/components/atoms/_toast.scss]: Fixed!

File: src/components/atoms/_toast.scss (hunk 3 above)
Comment (axeleriksson147) [reply src/components/atoms/_toast.scss]: Removed!

File: src/components/atoms/_toast.scss (hunk 4 above)
Comment (axeleriksson147) [reply src/components/atoms/_toast.scss]: Makes sense. Updated!

File: src/components/atoms/_toast.scss (hunk 1 above)
Comment (axeleriksson147) [reply src/components/atoms/_toast.scss:2]: I took the colors defined in _statics.scss that are pretty much the same as defined in the Figma. Maybe in another PR all colors should be aligned with what's defined in SDS.

File: docs/getting-started.md [hunk 5]
```diff
@@ -37,7 +37,7 @@ In some cases (depending on your bundling solution) you might need to indicate t
 
 ## Link
 
-To work locally with uitoolkit-styles, please change the following value
+To work locally with uitoolkit-styles, e.g. `yarn link`, please change the following value
```
Comment (axeleriksson147) [inline docs/getting-started.md:40]: Makes it easier to find while searching in documentation

File: src/variables/_z-indexes.scss [hunk 6]
```diff
@@ -3,5 +3,6 @@ $z-index-badge: 2;
 $z-index-input: 3;
 $z-index-tooltip: 4;
 $z-index-dropdown-menu: 5;
+$z-index-toast: 6;
```
Comment (axeleriksson147) [inline src/variables/_z-indexes.scss:6]: Seems like it should be above the ones listed cause it's an important message.

**Comment by axeleriksson147** [comment]
Changed to 8px border-radius

File: src/components/atoms/_toast.scss [hunk 7]
```diff
@@ -0,0 +1,69 @@
...
+    }
+
+    &__horizontal-center {
+        left: 50%;
+        position: absolute;
+        transform: translateX(-50%);
+    }
+
+    &__horizontal-left {
+        left: 16px;
```
Comment (symphony-jean-michael) [inline src/components/atoms/_toast.scss:34]: Replace by `toRem(16)`

File: src/components/atoms/_toast.scss [hunk 8]
```diff
@@ -0,0 +1,69 @@
...
+        position: absolute;
+    }
+
+    &__horizontal-right {
+        right: 16px;
```
Comment (symphony-jean-michael) [inline src/components/atoms/_toast.scss:39]: Same: `toRem(16)`

File: src/components/atoms/_toast.scss [hunk 9]
```diff
@@ -0,0 +1,69 @@
...
+        transform: translateY(-50%);
+    }
+    
+    &__vertical-top {
+        position: absolute;
+        top: 16px;
+    }
+
+    /*
+    * There's a special case when both horizontal and vertical is centered that needs to be harcdoded,
```
Comment (symphony-jean-michael) [inline src/components/atoms/_toast.scss:60]: Typo: harcdoded => hardcoded

File: src/components/atoms/_toast.scss [hunk 10]
```diff
@@ -0,0 +1,69 @@
...
+        position: absolute;
+    }
+
+    &__horizontal-right {
+        right: 16px;
+        position: absolute;
+    }
+
+    &__vertical-bottom {
+        bottom: 16px;
```
Comment (symphony-jean-michael) [inline src/components/atoms/_toast.scss:44]: Same `toRem(16)`

File: src/components/atoms/_toast.scss [hunk 11]
```diff
@@ -0,0 +1,69 @@
...
+
+    &__vertical-center {
+        position: absolute;
+        top: 50%;
+        transform: translateY(-50%);
+    }
+    
+    &__vertical-top {
+        position: absolute;
+        top: 16px;
```
Comment (symphony-jean-michael) [inline src/components/atoms/_toast.scss:56]: Same: `toRem(16)`

File: docs/getting-started.md (hunk 5 above)
Comment (symphony-jean-michael) [reply docs/getting-started.md:40]: 👍

File: src/components/atoms/_toast.scss (hunk 7 above)
Comment (axeleriksson147) [reply src/components/atoms/_toast.scss:34]: Thanks, fixed!

File: src/components/atoms/_toast.scss (hunk 9 above)
Comment (axeleriksson147) [reply src/components/atoms/_toast.scss]: Fixed!

**Review by symphony-jean-michael — APPROVED** [review]
LGTM :)

File: docs/getting-started.md (hunk 5 above)
Comment (AliciaSymphony) [reply docs/getting-started.md:40]: 🚀

**Review by AliciaSymphony — APPROVED** [review]
Looks good!

**Pull request merged**